dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -565,23 +575,75 @@ public static boolean versionIsGreaterThanCurrentlySupported(String semVerString
+            final boolean isVersionValid = wdlBridge.isVersionFieldValid(tempMainDescriptor.getAbsolutePath(), primaryDescriptorPath);
+            Optional<String> parsedVersionString = getSemanticVersionString(primaryDescriptorContent);
+
+            if (parsedVersionString.isEmpty()) {
+                if (isVersionValid) {
+                    // Return default version if there's no parsed 'version' field and the version is valid (no 'version' is valid)
+                    return Optional.of(DEFAULT_WDL_VERSION);
+                }
+                // If there's no parsed 'version' string and the version isn't valid, then there's likely something wrong with the version
+                return Optional.empty();
+            }
+            return parsedVersionString;
+        } catch (IOException e) {
+            LOG.error("Error creating temporary file for descriptor {}", primaryDescriptorPath);
+            throw new CustomWebApplicationException(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } finally {
+            FileUtils.deleteQuietly(tempMainDescriptor);
+        }
+    }
+
     /**
      * Get the semantic version string from the WDL file
-     * @param primaryDescriptorPath path to the primary WDL descriptor
+     * @param descriptorContent the file content of the primary WDL descriptor
      * @return the semantic version string, e.g. '1.0', which should be in the first code line, e.g. 'version 1.0' or 'draft-3'
      */
-    public static Optional<String> getSemanticVersionString(String primaryDescriptorPath) {
+    public static Optional<String> getSemanticVersionString(String descriptorContent) {
         WdlBridge wdlBridge = new WdlBridge();
-        Optional<String> firstCodeLine = wdlBridge.getFirstCodeLine(primaryDescriptorPath);
+        Optional<String> firstCodeLine = wdlBridge.getFirstCodeLine(descriptorContent);
+
         // https://www.scala-lang.org/files/archive/api/2.13.x/scala/jdk/javaapi/OptionConverters$.html
         // The WDL specification says that WDL descriptors from now on must have
         // a version string as the first line, e.g. 'version 1.0' or 'version draft-3'
         // https://github.com/openwdl/wdl/blob/main/versions/1.0/SPEC.md#versioning
         // however some very old WDL scripts may not have a version string line
         // Check to see if we found the first line of code and that it has two parts
         if (firstCodeLine.isPresent()) {
+            String wdlCommentSymbol = "#";
             String semanticVersionLine = firstCodeLine.get();
-            String[] semanticVersionStringArray = semanticVersionLine.split("\\s+");
+            // Remove comment in lines like 'version 1.0 # this is a comment'
+            String semanticVersionStringWithoutComments = semanticVersionLine.split(wdlCommentSymbol)[0];
+            String[] semanticVersionStringArray = semanticVersionStringWithoutComments.split("\\s+");
```

Comment:
In WDL, can a `version <x>` statement ever be indented with leading whitespace?  If so, there'll be a leading empty string in the array that results from the `split()`, and the following code will get confused.